test/inheritAttributes.test.js

```diff
@@ -490,3 +494,869 @@ QUnit.test('end to end - alternate BaseURLs', function(assert) {
+            mimeType= "video/mp6"
+            bandwidth= "5000000"
+            codecs= "avc1.64001e"
+            height= "404"
+            id= "test"
+            width= "720" >
+            <SegmentTemplate></SegmentTemplate>
+            <SegmentList></SegmentList>
+          </Representation>
+          <Representation
+            height= "545" >
+          </Representation>
+        </AdaptationSet>
+      </Period>
+    </MPD>
+  `
+)));
+
+  const expected = [{
+    attributes: {
+      bandwidth: '5000000',
+      baseUrl: 'https://www.example.com/base',
+      duration: 'PT0H4M40.414S',
+      codecs: 'avc1.64001e',
+      height: '404',
+      id: 'test',
+      lang: 'es',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'video/mp6',
+      periodIndex: 0,
+      role: {
+        value: 'main'
+      },
+      segmentAlignment: 'true',
+      sourceDuration: 30,
+      width: '720',
+      startWithSAP: '1'
+    },
+    segments: [
+      {
+        duration: 1,
+        map: {
+          uri: '',
+          resolvedUri: 'https://www.example.com/base'
+        },
+        resolvedUri: 'https://www.example.com/low/segment-1.ts',
+        timeline: 0,
+        uri: 'low/segment-1.ts'
+      },
+      {
+        duration: 1,
+        map: {
+          uri: '',
+          resolvedUri: 'https://www.example.com/base'
+        },
+        resolvedUri: 'https://www.example.com/low/segment-2.ts',
+        timeline: 0,
+        uri: 'low/segment-2.ts'
+      },
+      {
+        duration: 1,
+        map: {
+          uri: '',
+          resolvedUri: 'https://www.example.com/base'
+        },
+        resolvedUri: 'https://www.example.com/low/segment-3.ts',
+        timeline: 0,
+        uri: 'low/segment-3.ts'
+      },
+      {
+        duration: 1,
+        map: {
+          uri: '',
+          resolvedUri: 'https://www.example.com/base'
+        },
+        resolvedUri: 'https://www.example.com/low/segment-4.ts',
+        timeline: 0,
+        uri: 'low/segment-4.ts'
+      },
+      {
+        duration: 1,
+        map: {
+          uri: '',
+          resolvedUri: 'https://www.example.com/base'
+        },
+        resolvedUri: 'https://www.example.com/low/segment-5.ts',
+        timeline: 0,
+        uri: 'low/segment-5.ts'
+      },
+      {
+        duration: 1,
+        map: {
+          uri: '',
+          resolvedUri: 'https://www.example.com/base'
+        },
+        resolvedUri: 'https://www.example.com/low/segment-6.ts',
+        timeline: 0,
+        uri: 'low/segment-6.ts'
+      }
+    ]
+  }, {
+    attributes: {
+      baseUrl: 'https://www.example.com/base',
+      duration: 'PT0H4M40.414S',
+      lang: 'es',
+      height: '545',
+      mediaPresentationDuration: 'PT30S',
+      mimeType: 'video/mp4',
+      periodIndex: 0,
+      role: {
+        value: 'main'
+      },
+      segmentAlignment: 'true',
+      sourceDuration: 30,
+      startWithSAP: '1'
+
+    },
+    segments: undefined
```

Comment:
This should not be undefined, it should have a segment list built from the segment information provided in the adaptation set